components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3046,6 +3062,22 @@ private boolean isDomainSupportedAttribute(String attributeName) {
         }
     }
 
+    private String[] checkForSCIMDisabledHybridRoles(String[] roles) throws CharonException {
+
+        try {
+            SCIMGroupHandler groupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
+            Set<String> scimRoles = groupHandler.listSCIMRoles();
+            List<String> scimDisabledHybridRoles = getSCIMDisabledHybridRoleList(roles, scimRoles);
+            if (!scimDisabledHybridRoles.isEmpty()) {
+                createSCIMAttributesForSCIMDisabledHybridRoles(scimDisabledHybridRoles);
+            }
+        } catch (org.wso2.carbon.user.core.UserStoreException | IdentitySCIMException e) {
+            throw new CharonException("Error in retrieving SCIM Group information from database.", e);
+        }
+
+        return roles;
```

Comment:
What is the purpose of returning the roles as no modification is done to 'roles' within this method.

Reply:
Fixed